src/System.Runtime/tests/System/String.Split.cs

```diff
@@ -0,0 +1,515 @@
+        Assert.Equal(expected, value.Split(new[] { "," }, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, count, options));
+    }
+
+    [Fact]
+    public static void TestSplitOneCountSingleResult()
+    {
+        const string value = "a,b";
+        const int count = 1;
+        const StringSplitOptions options = StringSplitOptions.None;
+
+        string[] expected = new[] { value };
+
+        Assert.Equal(expected, value.Split(',', count, options));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count, options));
+        Assert.Equal(expected, value.Split(",", count, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, count, options));
+    }
+
+    [Fact]
+    public static void TestSplitNoMatchSingleResult()
+    {
+        const string value = "a b";
+        const int count = int.MaxValue;
+        const StringSplitOptions options = StringSplitOptions.None;
+
+        string[] expected = new[] { value };
+
+        Assert.Equal(expected, value.Split(','));
+        Assert.Equal(expected, value.Split(',', options));
+        Assert.Equal(expected, value.Split(',', count, options));
+        Assert.Equal(expected, value.Split(new[] { ',' }));
+        Assert.Equal(expected, value.Split(new[] { ',' }, options));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count));
+        Assert.Equal(expected, value.Split(new[] { ',' }, count, options));
+        Assert.Equal(expected, value.Split(","));
+        Assert.Equal(expected, value.Split(",", options));
+        Assert.Equal(expected, value.Split(",", count, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, options));
+        Assert.Equal(expected, value.Split(new[] { "," }, count, options));
+    }
+
+    private const int M = int.MaxValue;
+
+    [Theory]
+    [InlineData("", ',', 0, StringSplitOptions.None, new string[0])]
+    [InlineData("", ',', 1, StringSplitOptions.None, new[] { "" })]
+    [InlineData("", ',', 2, StringSplitOptions.None, new[] { "" })]
+    [InlineData("", ',', 3, StringSplitOptions.None, new[] { "" })]
+    [InlineData("", ',', 4, StringSplitOptions.None, new[] { "" })]
+    [InlineData("", ',', M, StringSplitOptions.None, new[] { "" })]
+    [InlineData("", ',', 0, StringSplitOptions.RemoveEmptyEntries, new string[0])]
+    [InlineData("", ',', 1, StringSplitOptions.RemoveEmptyEntries, new string[0])]
+    [InlineData("", ',', 2, StringSplitOptions.RemoveEmptyEntries, new string[0])]
+    [InlineData("", ',', 3, StringSplitOptions.RemoveEmptyEntries, new string[0])]
+    [InlineData("", ',', 4, StringSplitOptions.RemoveEmptyEntries, new string[0])]
+    [InlineData("", ',', M, StringSplitOptions.RemoveEmptyEntries, new string[0])]
+    [InlineData(",", ',', 0, StringSplitOptions.None, new string[0])]
+    [InlineData(",", ',', 1, StringSplitOptions.None, new[] { "," })]
```

Comment:
I obviously don't understand String.Split as well as I might have hoped. Why would we return a string containing the separator rather than empty string here?